zerodb/catalog/indexes/field.py

```diff
@@ -1,11 +1,58 @@
 import itertools as it
 
+from persistent import Persistent
+from ZODB.broken import Broken
 from repoze.catalog.indexes.field import CatalogFieldIndex as _CatalogFieldIndex
+from repoze.catalog.indexes.common import CatalogIndex
+from repoze.catalog import RangeValue
 from zerodb import trees
 from zerodb.catalog.indexes.common import CallableDiscriminatorMixin
 from zerodb.storage import prefetch
 from zerodb.util.iter import ListPrefetch
 
+import BTrees
+
+BTreeSet = BTrees.family32.IF.TreeSet
+
+_marker = ()
+
+threshold = 10
+
+
+def multiunion1(set_type, seqs):
+    # XXX simple/slow implementation. Goal is just to get tests to pass.
+    result = set_type()
+    for s in seqs:
+        try:
+            iter(s)
```

Comment:
Maybe we can just check: if the type isn't tuple, Bucket or TreeSet (or whatever multiunion takes) - treat it as a single value